exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/okhttp/OkHttpExporterBuilder.java

```diff
@@ -137,6 +147,21 @@ public OkHttpExporter<T> build() {
       clientBuilder.addInterceptor(new RetryInterceptor(retryPolicy, OkHttpExporter::isRetryable));
     }
 
+    if (authenticator != null) {
+      Authenticator finalAuthenticator = authenticator;
+      // Generate and attach OkHttp Authenticator implementation
+      clientBuilder.authenticator(
+          (Route route, Response rspns) -> {
+            Request.Builder requestBuilder = rspns.request().newBuilder();
+            finalAuthenticator.getHeaders(
```

Comment:
So do you still think it should be `getHeader(Consumer<Map<String, String>>)` or would `Map<String, String> getHeaders()` suffice?

Reply:
At least in the current form using the Consumer construct doesn't really add anything. And since the request needs the result before doing the follow up request the question is if it ever will make sense to have async in the interaction here. 

If we have the method return a CompletableFuture it would still need to do an instant get generally.

Perhaps have the getHeader simply return a Map<String, String> for now and when we consider the spec update we can consider if it makes sense to have it return a CompletableFuture or use a Consumer callback to make it more async capable.